### ISSUE TITLE
L026 error while using bigquery nested field

### ISSUE THREAD

**Issue by amardeep** [sqlfluff/sqlfluff#1503]
Consider the following sql:

```sql
WITH commit_info AS (
    SELECT
        commit,
        author.name AS author_name
    FROM `bigquery-public-data.github_repos.commits`
    LIMIT 10
)

SELECT * FROM commit_info
```

## Expected Behaviour
Lint should not fail with L026 error

## Observed Behaviour
Lint fails with following error:
```
L:   4 | P:   9 | L026 | Reference 'author.name' refers to table/view not found
                       | in the FROM clause or found in parent subquery.
```

## Dialect
bigquery

## Version
- sqlfluff, version 0.6.6
- Python 3.9.7

**Comment by tunetheweb** [comment]
I use BigQuery a lot and to be honest I turn off L026 in my `.sqlfluff` config as don't think this rule useful for BigQuery.

Perhaps there's a case to be made to exclude BigQuery from this rule by default? We did a similar change for another rule for postgres in #1482 but that would mean no BigQuery user can benefit from this rule even if they want it (i.e. they don't use structs) so not sure that's a good solution.

Honestly I think configuring it off is the best option.

**Comment by tunetheweb** [comment]
Decided to disable this rule (and also L028 for similar reasons) in #1504 for BigQuery. There is the option to force enable it for those that want it but agree default should be disabled.

**Comment by emmahunt** [comment]
We're experiencing this issue as well. Would be great if there could be some investigation into a solution, rather than disabling the rule 🙂 

CC: @jranks123
